### PR TITLE
Fixes to summit jsrun flags and mpi tests + correct debugging flags.

### PR DESCRIPTION
**Pull request by PDoakORNL**
Someone should approve this before merge.

### CODE REVIEW

**Comment by weilewei** [comment]
Hi Peter, I ran all the tests and got the following errors:
```
 53/146 Test  #53: mpi_collective_sum_test .....................***Failed    0.62 sec
122/146 Test #122: bilayer_lattice_Nc1_interband ...............***Failed    1.55 sec
123/146 Test #123: bilayer_lattice_Nc1_intra_plus_interband ....***Failed    1.55 sec
124/146 Test #124: bilayer_lattice_Nc1_intraband ...............***Failed    1.57 sec
125/146 Test #125: square_lattice_Nc4_nn .......................***Failed    1.56 sec
126/146 Test #126: square_lattice_Nc4_onSite_plus_nn ...........***Failed    1.54 sec
144/146 Test #144: dca_sp_DCA+_mpi_test ........................***Failed    1.58 sec
145/146 Test #145: dca_sp_DCA+_mpi_cuda_test ...................***Failed    1.60 sec
```
the error msgs for each failed test:
```
test 144
        Start 144: dca_sp_DCA+_mpi_test

144: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/system-level/dca/dca_sp_DCA+_mpi_test"
144: Test timeout computed to be: 1000
144: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
144/146 Test #144: dca_sp_DCA+_mpi_test ........................***Failed    1.58 sec
test 145
        Start 145: dca_sp_DCA+_mpi_cuda_test

145: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/system-level/dca/dca_sp_DCA+_mpi_cuda_test"
145: Test timeout computed to be: 1000
145: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
145/146 Test #145: dca_sp_DCA+_mpi_cuda_test ...................***Failed    1.60 sec
test 146
        Start 146: dca_sp_DCA+_thread_test

test 122
        Start 122: bilayer_lattice_Nc1_interband

122: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/integration/cluster_solver/ctaux/bilayer_lattice/Nc1_interband/bilayer_lattice_Nc1_interband"
122: Test timeout computed to be: 1000
122: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
122/146 Test #122: bilayer_lattice_Nc1_interband ...............***Failed    1.55 sec
test 123
        Start 123: bilayer_lattice_Nc1_intra_plus_interband

123: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/integration/cluster_solver/ctaux/bilayer_lattice/Nc1_intra_plus_interband/bilayer_lattice_Nc1_intra_plus_interband"
123: Test timeout computed to be: 1000
123: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
123/146 Test #123: bilayer_lattice_Nc1_intra_plus_interband ....***Failed    1.55 sec
test 124
        Start 124: bilayer_lattice_Nc1_intraband

124: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/integration/cluster_solver/ctaux/bilayer_lattice/Nc1_intraband/bilayer_lattice_Nc1_intraband"
124: Test timeout computed to be: 1000
124: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
124/146 Test #124: bilayer_lattice_Nc1_intraband ...............***Failed    1.57 sec
test 125
        Start 125: square_lattice_Nc4_nn

125: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/integration/cluster_solver/ctaux/square_lattice/Nc4_nn/square_lattice_Nc4_nn"
125: Test timeout computed to be: 1000
125: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
125/146 Test #125: square_lattice_Nc4_nn .......................***Failed    1.56 sec
test 126
        Start 126: square_lattice_Nc4_onSite_plus_nn

126: Test command: /sw/summit/xalt/1.1.4/bin/jsrun "-n" "8" "-a" "1" "-g" "1" "-c" "7" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/tmp/DCA/build/test/integration/cluster_solver/ctaux/square_lattice/Nc4_onSite_plus_nn/square_lattice_Nc4_onSite_plus_nn"
126: Test timeout computed to be: 1000
126: Resources unavailable:  CPU: avail 42 requested 56 [ slots 8 x cpus_per_rs 7 ]
126/146 Test #126: square_lattice_Nc4_onSite_plus_nn ...........***Failed    1.54 sec
test 127
        Start 127: NiO_no_change_test
```

suggested improvement:
For me, I tried to make `-n = 6` for these failed tests, but their test assertion failed as the result is within the acceptable range. Maybe you could look into each test in more detail to change mpi ranks in the test file. Or whenever `-n == 8`, make `-c==6` as total cores available on one Summit node is 42.

**Comment by PDoakORNL** [comment]
Yes I ran a two node run I think... Will work on a bit more.

File: cmake/dca_compiler_options.cmake [hunk 1]
```diff
@@ -41,6 +41,11 @@ if (DCA_HAVE_CUDA)
       -Xcompiler -Wno-unused-parameter
       -Xcompiler -Wno-switch
       -Xcompiler ${DCA_THREADING_FLAGS})
+    if (CMAKE_BUILD_TYPE MATCHES "Debug")
+      list(APPEND CUDA_NVCC_FALGS
```
Comment (gbalduzz) [inline cmake/dca_compiler_options.cmake]: Typo. Does the code still work if the user passed the same flag to the cached CUDA_NVCC_FLAGS variable?

**Comment by gbalduzz** [comment]
There seems to be some problem with the compiler flags.
I get: `cc: error: unrecognized command line option ‘-Xcompiler’; did you mean ‘--compile’?`
I agree with the changes to the MPI flags, if they also work on Daint (we need some if else otherwise), but do we need any of the CUDA flags besides maybe -G on debug?

**Comment by PDoakORNL** [comment]
fixed by other means

**Pull request closed without merging**